### DISCUSSION TITLE
Make use of solution filters

### DISCUSSION THREAD

**Discussion by nebula2** [abpframework/abp#14270] [Ideas]
Hey everyone. Just a random thought:

abp's code base increases in the amount of projects that reside in the .sln files.
Opening such a solution it can be time consuming and you may load way more projects than you need.

Using .slnf Files could be a good fit for this.

Docs: https://learn.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2022

a solution filter file is rather simple and may look like this:

`my-solution-filter.slnf`
```json
{
  "solution": {
    "path": "my-solution.sln",
    "projects": [
      "apps\\App1\\App1.csproj",
      "modules\\Module1\\Module1.csproj",
      "modules\\Module2\\Module2.csproj",
      "modules\\Module3\\Module3.csproj"
    ]
  }
}
```

also this works: `dotnet build my-solution-filter.slnf`

**Comment by maliming** [comment]
hi

Thanks for your advice, It seems that the feature does not support all IDEs(Rider).

![image](https://user-images.githubusercontent.com/6908465/194751691-26e082b0-2c44-4902-a368-2e6fa954d2b3.png)

**Reply by nebula2** [reply]
> hi
> 
> Thanks for your advice, It seems that the feature does not support all IDEs(Rider).
> 
> ![image](https://user-images.githubusercontent.com/6908465/194751691-26e082b0-2c44-4902-a368-2e6fa954d2b3.png)

[NotNull] annotation from JetBrains is also used, through only supported by Rider (as far as I know)
